### ISSUE TITLE
10000 DOGE reward. Unable to recover my private keys. I know the password of wallet.dat

### ISSUE THREAD

**Issue by usefulcat1546** [dogecoin/dogecoin#1876]
Hello, thank you for reading me.

If you manage to help me 10,000 dogecoin will be transferred to you with pleasure.

DM your dogecoin address.

**PROBLEM:**

I have around 1500 dogecoin addresses in my wallet.dat.

I know the password.

Dogecoin QT says: _"Error unlocking wallet: some keys decrypt but not all. Your wallet file may be corrupt. (Code -1)"_

In pywallet.py it tells me the password is incorrect.

I modified the source code of Dogecoin and Pywallet to unlock the wallet.

Only the first 100 keys are decrypted using the password.

I tried a recovery by HDseed / HDmaster (new and innactive) by generating 50000 addresses with the derivation _m/0'/0'_ BIP32 seen in wallet.cpp.

It gives me the first 100 private keys of the wallet. _( m/0'/0'/100' )_

**OBSERVATION:**
DEBUG.log

_Encrypting Wallet with an nDeriveIterations of 67515.

Password hased 151139 times

CWallet::NewKeyPool wrote 100 new keys_

In pywallet it gives me:

_nDerivationIterations: 151139._

I of course tried salvagewallet, zapwallettxes with copies of the wallet.

**UPDATE**

I had an old "test" wallet in the ".dogecoin" folder.
Strangely, the decrypted addresses are those of this wallet. It seems that the two wallets are intertwined.

the problem is that the masterkey, salt and nderiv are those of the old "test" wallet

I don't know where the new keys are stored on the wallet.

I did a search with a HEX editor; the only "mkey" is that of "test" wallet

**Comment by shibe2** [comment]
You may need to generate keys for both receiving and change addresses from seed. Then check balances and transaction history of all these addresses. If balances add up to expected amount of dogecoins, you can use the corresponding keys to access them. Otherwise further investigation will be needed based on transaction history of addresses,

**Comment by slightlyskepticalpotat** [comment]
Seconded. Since it sounds like you are comfortable working with Python, I would suggest you write a quick script to query the [chain.so API](https://chain.so/api) to quickly determine how much Dogecoin is in those addresses and if it matches your expectations before trying another way.

**Comment by usefulcat1546** [comment]
> You may need to generate keys for both receiving and change addresses from seed. Then check balances and transaction history of all these addresses. If balances add up to expected amount of dogecoins, you can use the corresponding keys to access them. Otherwise further investigation will be needed based on transaction history of addresses,

Thank you for your interest in this issue.  Should I attempt to generate addresses from the first 100 addresses successfully decrypt (change and receive).  Otherwise I have already generated 50,000 derivations with the addresses hdseed and inactivehdseed.  Only the first 100 addresses match.  I conclude that the 100 addresses generated with hdseed were generated after encryption.  The one with inactivehdseed including those during the creation of the wallet.  If I don't make a mistake.

**Comment by usefulcat1546** [comment]
> Seconded. Since it sounds like you are comfortable working with Python, I would suggest you write a quick script to query the [chain.so API](https://chain.so/api) to quickly determine how much Dogecoin is in those addresses and if it matches your expectations before trying another way.

Hi
Thank you for your reply.

By doing the dump with pywallet, I was able to get the list of all the addresses in the wallet.  Several addresses contain small amounts.  Only one contains a large amount of dogecoin.  So I searched in the keys generated with the derivations path m/0'/0' of hdseed and inactivehdseed and allowed 50000 addresses generate, the address not listed there.

Thank you again

**Comment by TopZhu** [comment]
Hi, I think you can use this script to extract your addresses and private keys:
https://github.com/akx/walletool

**Comment by usefulcat1546** [comment]
During my investigations, I noticed that the Master Key only decrypts one part of the wallet. The addresses generated at creation.

 The HDseed also only generates the same addresses.  

I have a backup of the portfolio from the beginning.  Pywallet cannot open it and Dogecoin Darmon crashes when opening.  "Unexpected end of file".  

I have confirmation that the other addresses are indeed derived from an HDSEED.  I tried a dump with bitcoin-wallet. He give me the public key, the encrypted private key and a 64 character string that does not match the decrypted private key. 

I'm in a bind ...

**Comment by shibe2** [comment]
How do you think the wallet might got damaged?

**Comment by usefulcat1546** [comment]
I have expolorized the data of Wallet.dat, I have been able to find unencrypted private keys of some derivation.
I was also able to retrieve the address and the public key used for the masterkey that interests me but not the private key.
There are a total of 4 master keys in the wallet.
As soon as I finish studying this problem I will submit a pull request.

I have several information:
Private, public and encrypted keys for several addresses.
The address and public key of the masterkey.

It seems that the keys have been encrypted with a different masterkey, salt and deriation than the mkey line of the wallet.

Based on this information you have an idea of how to recover the funds. The bonus of 1000 DOGE is still to be taken.

**Comment by usefulcat1546** [comment]
Hi friend,
I ran a script to scan all the hax sequences and considered them as private keys, the script converted every 64 character sequence as a private key and then converted it to a Dogecoin address. If an address matches one of mine, an alert is created.
I give the script in case someone needs it:

https://github.com/usefulcat1546/hex-scan-wallet-dat

There are more than 4500000 characters so my intel I7 will take a few days, in the meantime I try to locate in the wallet if there is a line which could be another masterkey and salt, since I know the word of pass and derivation.

**Comment by shibe2** [comment]
I have similar program in C++. It only works for unencrypted keys.

**Comment by usefulcat1546** [comment]
The scan did not give anything, there are indeed two wallets intertwined in one, and the decryption keys only correspond to the first wallet. My funds seem lost. If a developer goes through it, it will be a good idea to find the source of the problem and come up with a fix.

**Comment by nmanpkr** [comment]
Did you try to use pywallet --recover? Create a new drive partition and copy the wallet.dat onto it, then run pywallet --recover --recov_device /dev/sd## --recov_size #blocksize --recov_outputdir /home/name/

**Comment by shibe2** [comment]
So how do you think your wallet might have ended up in this state?

**Comment by coltwill2121** [comment]
sounds stupid and it probly is.. but try printing a paper wallet.. it will give you private keys..im a newbie hope this helps

**Comment by patricklodder** [comment]
> sounds stupid and it probly is.. but try printing a paper wallet.. it will give you private keys..im a newbie hope this helps

Private keys for paper wallets are generated on the fly and are not stored in your wallet file, so this does not work.

**Comment by usefulcat1546** [comment]
Yes Indeed, wallet file is different. I still haven't found a solution to this problem, I hope an expert in cryptography and crypto programming will be kind enough to look into the problem. In any case, the reward of 10,000 DOGE for the person who manages to help me with it will remain available. Thank you all for your help. I will give you to all those who have helped me even without success a reward.

**Comment by usefulcat1546** [comment]
> Did you try to use pywallet --recover? Create a new drive partition and copy the wallet.dat onto it, then run pywallet --recover --recov_device /dev/sd## --recov_size #blocksize --recov_outputdir /home/name/

Hi
Yes, already try and this found the same result :/

**Comment by DemonRx** [comment]
Just a thought, did you try to do data recovery of the original storage medium the wallet file was on?

**Comment by coltwill2121** [comment]
hope this helps. again remember i am still a newb .. i was playing with the console in dogecoin core and thought maybe you could play with some command lines.. but im sure you have already do this... 

02:39:16
￼
Welcome to the Dogecoin Core RPC console.
Use up and down arrows to navigate history, and Ctrl-L to clear screen.
Type help for an overview of available commands.
WARNING: Scammers have been active, telling users to type commands here, stealing their wallet contents. Do not use this console without fully understanding the ramification of a command.


02:42:01
￼
salvagewallet


02:42:01
￼
Method not found (code -32601)


02:44:00
￼
salvagewallet


02:44:00
￼
Method not found (code -32601)


02:44:25
￼
help


02:44:25
￼
== Blockchain ==
getbestblockhash
getblock "blockhash" ( verbose )
getblockchaininfo
getblockcount
getblockhash height
getblockheader "hash" ( verbose )
getchaintips
getdifficulty
getmempoolancestors txid (verbose)
getmempooldescendants txid (verbose)
getmempoolentry txid
getmempoolinfo
getrawmempool ( verbose )
gettxout "txid" n ( include_mempool )
gettxoutproof ["txid",...] ( blockhash )
gettxoutsetinfo
preciousblock "blockhash"
pruneblockchain
verifychain ( checklevel nblocks )
verifytxoutproof "proof"

== Control ==
getinfo
getmemoryinfo
help ( "command" )
stop

== Generating ==
generate nblocks ( maxtries )
generatetoaddress nblocks address (maxtries)

== Mining ==
getauxblock (hash auxpow)
getblocktemplate ( TemplateRequest )
getmininginfo
getnetworkhashps ( nblocks height )
prioritisetransaction <txid> <priority delta> <fee delta>
submitblock "hexdata" ( "jsonparametersobject" )

== Network ==
addnode "node" "add|remove|onetry"
clearbanned
disconnectnode "address" 
getaddednodeinfo ( "node" )
getconnectioncount
getnettotals
getnetworkinfo
getpeerinfo
listbanned
ping
setban "subnet" "add|remove" (bantime) (absolute)
setnetworkactive true|false

== Rawtransactions ==
createrawtransaction [{"txid":"id","vout":n},...] {"address":amount,"data":"hex",...} ( locktime )
decoderawtransaction "hexstring"
decodescript "hexstring"
fundrawtransaction "hexstring" ( options )
getrawtransaction "txid" ( verbose )
sendrawtransaction "hexstring" ( allowhighfees )
signrawtransaction "hexstring" ( [{"txid":"id","vout":n,"scriptPubKey":"hex","redeemScript":"hex"},...] ["privatekey1",...] sighashtype )

== Util ==
createmultisig nrequired ["key",...]
estimatefee nblocks
estimatepriority nblocks
estimatesmartfee nblocks
estimatesmartpriority nblocks
signmessagewithprivkey "privkey" "message"
validateaddress "address"
verifymessage "address" "signature" "message"

== Wallet ==
abandontransaction "txid"
addmultisigaddress nrequired ["key",...] ( "account" )
addwitnessaddress "address"
backupwallet "destination"
bumpfee "txid" ( options ) 
dumpprivkey "address"
dumpwallet "filename"
getaccount "address"
getaccountaddress "account"
getaddressesbyaccount "account"
getbalance ( "account" minconf include_watchonly )
getnewaddress ( "account" )
getrawchangeaddress
getreceivedbyaccount "account" ( minconf )
getreceivedbyaddress "address" ( minconf )
gettransaction "txid" ( include_watchonly )
getunconfirmedbalance
getwalletinfo
importaddress "address" ( "label" rescan p2sh )
importmulti "requests" "options"
importprivkey "bitcoinprivkey" ( "label" ) ( rescan )
importprunedfunds
importpubkey "pubkey" ( "label" rescan )
importwallet "filename"
keypoolrefill ( newsize )
listaccounts ( minconf include_watchonly)
listaddressgroupings
listlockunspent
listreceivedbyaccount ( minconf include_empty include_watchonly)
listreceivedbyaddress ( minconf include_empty include_watchonly)
listsinceblock ( "blockhash" target_confirmations include_watchonly)
listtransactions ( "account" count skip include_watchonly)
listunspent ( minconf maxconf  ["addresses",...] [include_unsafe] [query_options])
lockunspent unlock ([{"txid":"txid","vout":n},...])
move "fromaccount" "toaccount" amount ( minconf "comment" )
removeprunedfunds "txid"
sendfrom "fromaccount" "toaddress" amount ( minconf "comment" "comment_to" )
sendmany "fromaccount" {"address":amount,...} ( minconf "comment" ["address",...] )
sendtoaddress "address" amount ( "comment" "comment_to" subtractfeefromamount )
setaccount "address" "account"
settxfee amount
signmessage "address" "message"
walletlock
walletpassphrase "passphrase" timeout
walletpassphrasechange "oldpassphrase" "newpassphrase"


02:45:25
￼
dumpwallet


02:45:25
￼
dumpwallet "filename"

Dumps all wallet keys in a human-readable format.

Arguments:
1. "filename"    (string, required) The filename

Examples:
> dogecoin-cli dumpwallet "test"
> curl --user myusername --data-binary '{"jsonrpc": "1.0", "id":"curltest", "method": "dumpwallet", "params": ["test"] }' -H 'content-type: text/plain;' http://127.0.0.1:8332/
 (code -1)


02:49:20
￼
importpubkey 


02:49:20
￼
importpubkey "pubkey" ( "label" rescan )

Adds a public key (in hex) that can be watched as if it were in your wallet but cannot be used to spend.

Arguments:
1. "pubkey"           (string, required) The hex-encoded public key
2. "label"            (string, optional, default="") An optional label
3. rescan               (boolean, optional, default=true) Rescan the wallet for transactions

Note: This call can take minutes to complete if rescan is true.

Examples:

Import a public key with rescan
> dogecoin-cli importpubkey "mypubkey"

Import using a label without rescan
> dogecoin-cli importpubkey "mypubkey" "testing" false

As a JSON-RPC call
> curl --user myusername --data-binary '{"jsonrpc": "1.0", "id":"curltest", "method": "importpubkey", "params": ["mypubkey", "testing", false] }' -H 'content-type: text/plain;' http://127.0.0.1:8332/
 (code -1)

**Comment by coltwill2121** [comment]
just one more thought.. could you delete the test wallet ? maybe doing that would fix the intertwined

**Comment by shibe2** [comment]
Did you have the main wallet in non-default location?

**Comment by usefulcat1546** [comment]
I have in possession the test wallet with the first 100 addresses generated. I also have the blocked wallet with over 2000 addresses including the 100 test wallet addresses. 
The mkey part in this wallet corresponds to the test wallet.

**Comment by usefulcat1546** [comment]
I'll be ready to give 30,000 Dogecoin to whoever unlocks it for me. I also reward the people who have already helped me. thanks a lot

**Comment by shibe2** [comment]
I guess, you may have accidentally replaced the wallet.dat while Dogecoin Core was running.

**Comment by ReverseControl** [comment]
@usefulcat1546 What is the status of this issue?

**Comment by usefulcat1546** [comment]
> I guess, you may have accidentally replaced the wallet.dat while Dogecoin Core was running.

I know that the two Wallets were in the same folder and that the rpctimeout command surely played a role in the conflict between the wallets

**Comment by usefulcat1546** [comment]
> @usefulcat1546 What is the status of this issue?

The 30,000 Dogecoin reward is still available, I have over 320,000 doge blocked. Thank you

**Comment by ReverseControl** [comment]
@usefulcat1546 Which dogecoin core version did you create and encrypt the wallets with?

**Comment by usefulcat1546** [comment]
> @usefulcat1546 Which dogecoin core version did you create and encrypt the wallets with?

With the version 1.10, The latest version available in February 2019.

**Comment by shahroozam** [comment]
D6ftB9SDo9w1i4mT6Hb8Km3xaNj4vsjjCN

**Comment by ReverseControl** [comment]
@usefulcat1546 As this is not an issue that pertains to the Dogecoin Core Development I am closing this issue. However, you can ask the community over at discussions (https://github.com/dogecoin/dogecoin/discussions).

**Comment by usefulcat1546** [comment]
> @usefulcat1546 As this is not an issue that pertains to the Dogecoin Core Development I am closing this issue. However, you can ask the community over at discussions (https://github.com/dogecoin/dogecoin/discussions). 

This error is not my control and there is a real problem in the source code, this is not supposed to happen. developers should look into the problem so that it does not happen again.

**Comment by ReverseControl** [comment]
@usefulcat1546 Per your report, it seems your wallet is indeed corrupted. Not something we the developers would address. To properly help you recover your coins a developer would need access to your wallet, or a lot of patience for back and forth discussion; as has happened here already. I suggested the discussions section as that is more suitable for your situation. I do hope you are able to recover your coins. Please start a discussion on this matter, I am sure there are plenty of technical folks would be willing to put it in the time to recover your coins for a suitable reward.

**Comment by rookie33** [comment]
Have you tried backing up the wallet and -salvagewallet? It may resolve your issue. However, since it will take a long time to recover if the wallet is synced - remove the blockchain files from the AppData folder, then start the wallet with -salvagewallet and when it loads - close it, then copy/paste the blockchain files back, and reset the wallet.

**Comment by pythonboy20** [comment]
@usefulcat1546 
did you het back your dogecoin wallet? I can solve your problem
so if you want help to get your lost doge wallet, send me email:
[[anywalletrecovery@gmail.com]

**Comment by Petermaldonado** [comment]
Hi... Did you manage to solve it?.  If you still haven't figured it out, I think I can help you.  I work in forensics, and we're used to this kind of situation.  If you want send me a copy of the corrupted wallet, and within 5 days I will send you the results.  Email peterfmaldonado@gmail.com.
Ola... Conseguiu resolver?. Se vc ainda não conseguiu, creio que eu possa te ajudar. Trabalho na perícia criminal, e estamos acostumados com este tipo de situação. Se vc quiser me mande uma cópia da carteira corrompida, e no máximo 5 dias lhe envio os resultados. Email peterfmaldonado@gmail.com.

**Comment by keychainx** [comment]
My 5 cents to this discussion is that its not a pure Doge Core problem. I know at least 2 other similar cases where the "correct" password was only decrypting "half" of the keys from Bitcoin Core Pre 2018. The other case is a person moved the wallet.dat while Core was open and only partial keys showed up, even though the transaction is inside the wallet.dat file...

**Comment by arthurwolf** [comment]
@usefulcat1546 still looking for help with this?

**Comment by keychainx** [comment]
I can try to dump it for you

But would need the wallet.dat and the password to check

Sent with [Proton Mail](https://proton.me/) secure email.

------- Original Message -------

**Comment by coltwill2121** [comment]
long shot but maybe... maybe go to all settings.. click on search- find my
files, permissions (searching windows) look and see if you have files that
are in the exclude list.. if sooo.. remove them and you just might find
what you are looking for!!! my fingers are crossed!

**Comment by arthurwolf** [comment]
@usefulcat1546 you really shouldn't send your wallet/passwords to anyone for any reason.

people can help you by writing tools that you can execute yourself on airgapped computers

if you can clearly define what you are looking for in the file, I can write tools like that for you, no issue (as html/js, so super easy to run, just open the file in a browser)

**Comment by CryptoUnchained** [comment]
Did this happen to be installed on PC before upgrade to Win8/10 or 11? Or did you use chrome browser/android app? If so I can tell you where they've moved and files you can typically use master wallet pk or import to new wallet; make paper wallets reset pw etc; what you are describing looks like something I've seen and fixed before and has various issues related to which version Doge; win; and other changes that made a lot of wallets seem broken if not maintained updated regularly (or required functions denied by windows****:google that need some specific changes to work IMO/IME idk I successfully recovered 2014 Doge wallet wallet installed originally on XP and never upgraded and ran into a number of issues you describe along way before finding what I was missing

**Comment by usefulcat1546** [comment]
At the moment I have this information for some address : 
-Private Key
-Public Key
-Crypted Private Key
-Derivation Path

I tried to know masterkey from IV cipher and plain text for AES-256-CBC but that is not possible.

I'm looking for a way to find the masterkey from a child key but it also seems impossible

I am scanning the wallet to find out if the masterkey will not be in the wallet

I found 7 unencrypted private keys in the wallet

**Comment by usefulcat1546** [comment]
> Did this happen to be installed on PC before upgrade to Win8/10 or 11? Or did you use chrome browser/android app? If so I can tell you where they've moved and files you can typically use master wallet pk or import to new wallet; make paper wallets reset pw etc; what you are describing looks like something I've seen and fixed before and has various issues related to which version Doge; win; and other changes that made a lot of wallets seem broken if not maintained updated regularly (or required functions denied by windows****:google that need some specific changes to work IMO/IME idk I successfully recovered 2014 Doge wallet wallet installed originally on XP and never upgraded and ran into a number of issues you describe along way before finding what I was missing

This happened on an Ubuntu 18.04 LTS server. I could only retrieve the wallet.dat and the debug.log which indicates the derivation only

**Comment by usefulcat1546** [comment]
> @usefulcat1546 you really shouldn't send your wallet/passwords to anyone for any reason.
> 
> people can help you by writing tools that you can execute yourself on airgapped computers
> 
> if you can clearly define what you are looking for in the file, I can write tools like that for you, no issue (as html/js, so super easy to run, just open the file in a browser)

Thanks Arthur, Yes I am simply looking for a decryption algorithm in python nodejs c++ or whatever I can execute or a hexadecimal character string that I can locate in the wallet

**Comment by josephlarry** [comment]
I was an early adopter of digital currency, but my first move was wrong and nearly shade me off, I met a guy on Telegram who claimed to have been a Forex and Trading cryptocurrency expert, he lured me into investing in his company, I did, I invested $50000, and was promised profits. When its due time profits, I figured I was been scammed, I had to find a way out, Luckily I was introduced to a nerd ( ACCESSRECOVERYCONSULTANT AT GMAIL COM ) who later helped me recover my bitcoin from the company. I’d suggest you reach him for recovering of your digital assets.

**Comment by georgeartem** [comment]
Interested. Feel free to reach out 👍Sent from my iPhoneOn Apr 23, 2024, at 7:22 PM, josephlarry ***@***.***> wrote:﻿
I was an early adopter of digital currency, but my first move was wrong and nearly shade me off, I met a guy on Telegram who claimed to have been a Forex and Trading cryptocurrency expert, he lured me into investing in his company, I did, I invested $50000, and was promised profits. When its due time profits, I figured I was been scammed, I had to find a way out, Luckily I was introduced to a nerd ( ACCESSRECOVERYCONSULTANT AT GMAIL COM ) who later helped me recover my bitcoin from the company. I’d suggest you reach him for recovering of your digital assets.
